File: services/product/src/main/java/com/ecommerce/variant/purchase/PurchaseResponse.java
package com.ecommerce.variant.purchase;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.Map;

@Builder
public record PurchaseResponse(
        Long productId,
        Long variantId,
        String name,
        String primaryImagePath,
        BigDecimal price,
        int quantity,
        int availableQuantity,
        boolean isAvailable,
        BigDecimal totalPrice,
        Map<String, String> values
) {}

File: store-ui/src/app/services/models/order/purchase-response.ts
export interface PurchaseResponse {
    productId?: number;
    variantId?: number;
    name?: string;
    primaryImagePath?: string;
    price?: number;
    quantity?: number;
    availableQuantity?: number;
    isAvailable?: number;
    totalPrice?: number;
    values?: Record<string, string>;
}
